<div style=" background-color: RGB(0,114,200);" >
<h1 style="margin: auto; padding: 20px 0; color:#fff; text-align: center">PROJET 4 DATA ANALYST</h1>
<h2 style="margin: auto; padding: 20px 0; color:#fff; text-align: center">Réalisez une étude de santé publique avec R ou Python
</h2>
</div>

# OBJECTIF DE CE NOTEBOOK

Bienvenue dans l'outil plébiscité par les analystes de données Jupyter.

Il s'agit d'un outil permettant de mixer et d'alterner codes, textes et graphique.

Cet outil est formidable pour plusieurs raisons:

+ il permet de tester des lignes de codes au fur et à mesure de votre rédaction, de constater immédiatement le résultat d'un instruction, de la corriger si nécessaire.
+ De rédiger du texte pour expliquer l'approche suivie ou les résultats d'une analyse et de le mettre en forme grâce à du code html ou plus simple avec **Markdown**
+ d'agrémenter de graphiques

Pour vous aider dans vos premiers pas à l'usage de Jupyter et de Python, nous avons rédigé ce notebook en vous indiquant les instructions à suivre.

Il vous suffit pour cela de saisir le code Python répondant à l'instruction donnée.

Vous verrez de temps à autre le code Python répondant à une instruction donnée mais cela est fait pour vous aider à comprendre la nature du travail qui vous est demandée.

Et garder à l'esprit, qu'il n'y a pas de solution unique pour résoudre un problème et qu'il y a autant de résolutions de problèmes que de développeurs ;)...



<div style="background-color: RGB(0,150,250);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">Etape 1 - Importation des librairies et chargement des fichiers</h2>
</div>

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">1.1 - Importation des librairies</h3>
</div>

In [1]:
#Importation de la librairie Pandas
import pandas as pd

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">1.2 - Chargement des fichiers Excel</h3>
</div>

In [2]:
#Importation du fichier population.csv
population = pd.read_csv('population.csv')

#Importation du fichier dispo_alimentaire.csv
dispo_alimentaire = pd.read_csv('dispo_alimentaire.csv', sep=";")

#Importation du fichier aide_alimentaire.csv
aide_alimentaire = pd.read_csv('aide_alimentaire.csv')

#Importation du fichier sous_nutrition.csv
sous_nutrition = pd.read_csv('sous_nutrition.csv')


In [3]:
# Uniformisation de la dénomination de 5 pays

corrections_geo = {
    "Swaziland" : "Eswatini",
    "Chine- continentale" : "Chine, continentale",
    "Chine- Taiwan Province de" : "Chine, Taiwan Province de",
    "Royaume-Uni" : "Royaume-Uni de Grande-Bretagne et d'Irlande du Nord",
    "Tchéquie (la)" : "Tchéquie"
}

# Implantation des modifications dans les trois fichiers originaux
dispo_alimentaire['Zone'] = dispo_alimentaire['Zone'].replace(corrections_geo)
aide_alimentaire['Pays bénéficiaire'] = aide_alimentaire['Pays bénéficiaire'].replace(corrections_geo)
sous_nutrition['Zone'] = sous_nutrition['Zone'].replace(corrections_geo)

<div style="background-color: RGB(0,150,250);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">Etape 2 - Analyse exploratoire des fichiers</h2>
</div>

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.1 - Analyse exploratoire du fichier population</h3>
</div>

In [4]:
#Afficher les dimensions du dataset
print("Le tableau comporte {} observation(s) ou article(s)".format(population.shape[0]))
print("Le tableau comporte {} colonne(s)".format(population.shape[1]))

Le tableau comporte 1416 observation(s) ou article(s)
Le tableau comporte 3 colonne(s)


In [5]:
#Consulter le nombre de colonnes
#La nature des données dans chacune des colonnes
#Le nombre de valeurs présentes dans chacune des colonnes
print("Le nombre de colonnes est de {}".format(len(population.columns)))
print("Les types de donneés dans chaque colonne sont :\n{}".format(population.dtypes))
print("Pour chaque colonne, voici le nombre de valeurs :\n{}".format(population.count()))

Le nombre de colonnes est de 3
Les types de donneés dans chaque colonne sont :
Zone       object
Année       int64
Valeur    float64
dtype: object
Pour chaque colonne, voici le nombre de valeurs :
Zone      1416
Année     1416
Valeur    1416
dtype: int64


In [6]:
#Affichage les 5 premières lignes de la table
population.head()

,Zone,Année,Valeur
0,Afghanistan,2013,32269.589
1,Afghanistan,2014,33370.794
2,Afghanistan,2015,34413.603
3,Afghanistan,2016,35383.032
4,Afghanistan,2017,36296.113


In [7]:
#Nous allons harmoniser les unités. Pour cela, nous avons décidé de multiplier la population par 1000
#Multiplication de la colonne valeur par 1000
population['Valeur']=population['Valeur']*1000
population.head()

,Zone,Année,Valeur
0,Afghanistan,2013,32269589.0
1,Afghanistan,2014,33370794.0
2,Afghanistan,2015,34413603.0
3,Afghanistan,2016,35383032.0
4,Afghanistan,2017,36296113.0


In [8]:
#changement du nom de la colonne Valeur par Population
population=population.rename(columns={'Valeur': 'Population'})
population['Population'] = population['Population'].astype(int)

In [9]:
#Affichage les 5 premières lignes de la table pour voir les modifications
population.head()

,Zone,Année,Population
0,Afghanistan,2013,32269589
1,Afghanistan,2014,33370794
2,Afghanistan,2015,34413603
3,Afghanistan,2016,35383032
4,Afghanistan,2017,36296113


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.2 - Analyse exploratoire du fichier disponibilité alimentaire</h3>
</div>

In [10]:
#Afficher les dimensions du dataset
print("Le tableau comporte {} observation(s) ou article(s)".format(dispo_alimentaire.shape[0]))
print("Le tableau comporte {} colonne(s)".format(dispo_alimentaire.shape[1]))

Le tableau comporte 15605 observation(s) ou article(s)
Le tableau comporte 18 colonne(s)


In [11]:
#Consulter le nombre de colonnes
print("Le nombre de colonnes est de {}".format(len(dispo_alimentaire.columns)))

print("\nLes types de donneés dans chaque colonne sont :\n{}".format(dispo_alimentaire.dtypes))

print("\nPour chaque colonne, voici le nombre de valeurs :\n{}".format(dispo_alimentaire.count()))

Le nombre de colonnes est de 18

Les types de donneés dans chaque colonne sont :
Zone                                                              object
Produit                                                           object
Origine                                                           object
Disponibilité alimentaire (Kcal/personne/jour)                   float64
Disponibilité alimentaire en quantité (kg/personne/an)           float64
Disponibilité de matière grasse en quantité (g/personne/jour)    float64
Disponibilité de protéines en quantité (g/personne/jour)         float64
Disponibilité intérieure                                         float64
Exportations - Quantité                                          float64
Importations - Quantité                                          float64
Nourriture                                                       float64
Pertes                                                           float64
Production                                 

In [12]:
#Affichage les 5 premières lignes de la table
dispo_alimentaire.head()

,Zone,Produit,Origine,Disponibilité alimentaire (Kcal/personne/jour),Disponibilité alimentaire en quantité (kg/personne/an),Disponibilité de matière grasse en quantité (g/personne/jour),Disponibilité de protéines en quantité (g/personne/jour),Disponibilité intérieure,Exportations - Quantité,Importations - Quantité,Nourriture,Pertes,Production,Semences,Traitement,Aliments pour animaux,Autres Utilisations,Variation de stock
0,Afghanistan,Abats Comestible,animale,5.0,1.72,0.20,0.77,53.0,NaN,NaN,53.0,NaN,53.0,NaN,NaN,NaN,NaN,NaN
1,Afghanistan,Agrumes- Autres,vegetale,1.0,1.29,0.01,0.02,41.0,2.0,40.0,39.0,2.0,3.0,NaN,NaN,NaN,NaN,NaN
2,Afghanistan,Aliments pour enfants,vegetale,1.0,0.06,0.01,0.03,2.0,NaN,2.0,2.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,Afghanistan,Ananas,vegetale,0.0,0.00,NaN,NaN,0.0,NaN,0.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,Afghanistan,Bananes,vegetale,4.0,2.70,0.02,0.05,82.0,NaN,82.0,82.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [13]:
#remplacement des NaN dans le dataset par des 0
dispo_alimentaire = dispo_alimentaire.fillna(0)


In [14]:
#multiplication de toutes les lignes contenant des milliers de tonnes en Kg
colonnes_a_multiplier = ['Disponibilité intérieure', 'Exportations - Quantité', 'Importations - Quantité',
                         'Nourriture', 'Pertes', 'Production', 'Semences', 'Traitement',
                         'Aliments pour animaux', 'Autres Utilisations', 'Variation de stock']
dispo_alimentaire[colonnes_a_multiplier] = dispo_alimentaire[colonnes_a_multiplier].astype('int64')
dispo_alimentaire[colonnes_a_multiplier] = dispo_alimentaire[colonnes_a_multiplier] *1000000

In [15]:
#Affichage les 5 premières lignes de la table

dispo_alimentaire.head()

,Zone,Produit,Origine,Disponibilité alimentaire (Kcal/personne/jour),Disponibilité alimentaire en quantité (kg/personne/an),Disponibilité de matière grasse en quantité (g/personne/jour),Disponibilité de protéines en quantité (g/personne/jour),Disponibilité intérieure,Exportations - Quantité,Importations - Quantité,Nourriture,Pertes,Production,Semences,Traitement,Aliments pour animaux,Autres Utilisations,Variation de stock
0,Afghanistan,Abats Comestible,animale,5.0,1.72,0.20,0.77,53000000,0,0,53000000,0,53000000,0,0,0,0,0
1,Afghanistan,Agrumes- Autres,vegetale,1.0,1.29,0.01,0.02,41000000,2000000,40000000,39000000,2000000,3000000,0,0,0,0,0
2,Afghanistan,Aliments pour enfants,vegetale,1.0,0.06,0.01,0.03,2000000,0,2000000,2000000,0,0,0,0,0,0,0
3,Afghanistan,Ananas,vegetale,0.0,0.00,0.00,0.00,0,0,0,0,0,0,0,0,0,0,0
4,Afghanistan,Bananes,vegetale,4.0,2.70,0.02,0.05,82000000,0,82000000,82000000,0,0,0,0,0,0,0


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.3 - Analyse exploratoire du fichier aide alimentaire</h3>
</div>

In [16]:
#Afficher les dimensions du dataset
print("Le tableau comporte {} observation(s) ou article(s)".format(aide_alimentaire.shape[0]))
print("Le tableau comporte {} colonne(s)".format(aide_alimentaire.shape[1]))

Le tableau comporte 1475 observation(s) ou article(s)
Le tableau comporte 4 colonne(s)


In [17]:
#Consulter le nombre de colonnes
print("Le tableau comporte {} colonnes".format(len(aide_alimentaire.columns)))
print("Les types de donneés dans chaque colonne sont :\n{}".format(aide_alimentaire.dtypes))
print("Pour chaque colonne, voici le nombre de valeurs :\n{}".format(aide_alimentaire.count()))

Le tableau comporte 4 colonnes
Les types de donneés dans chaque colonne sont :
Pays bénéficiaire    object
Année                 int64
Produit              object
Valeur                int64
dtype: object
Pour chaque colonne, voici le nombre de valeurs :
Pays bénéficiaire    1475
Année                1475
Produit              1475
Valeur               1475
dtype: int64


In [18]:
#Affichage les 5 premières lignes de la table
aide_alimentaire.head()

,Pays bénéficiaire,Année,Produit,Valeur
0,Afghanistan,2013,Autres non-céréales,682
1,Afghanistan,2014,Autres non-céréales,335
2,Afghanistan,2013,Blé et Farin,39224
3,Afghanistan,2014,Blé et Farin,15160
4,Afghanistan,2013,Céréales,40504


In [19]:
#changement du nom de la colonne Pays bénéficiaire par Zone
aide_alimentaire=aide_alimentaire.rename(columns={'Pays bénéficiaire': 'Zone'})

In [20]:
#Multiplication de la colonne Aide_alimentaire qui contient des tonnes par 1000 pour avoir des kg
aide_alimentaire['Valeur']=aide_alimentaire['Valeur']*1000

In [21]:
#Affichage les 5 premières lignes de la table
aide_alimentaire.head()

,Zone,Année,Produit,Valeur
0,Afghanistan,2013,Autres non-céréales,682000
1,Afghanistan,2014,Autres non-céréales,335000
2,Afghanistan,2013,Blé et Farin,39224000
3,Afghanistan,2014,Blé et Farin,15160000
4,Afghanistan,2013,Céréales,40504000


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.3 - Analyse exploratoire du fichier sous nutrition</h3>
</div>

In [22]:
#Afficher les dimensions du dataset
print("Le tableau comporte {} observation(s) ou article(s)".format(sous_nutrition.shape[0]))
print("Le tableau comporte {} colonne(s)".format(sous_nutrition.shape[1]))

Le tableau comporte 1218 observation(s) ou article(s)
Le tableau comporte 3 colonne(s)


In [23]:
#Consulter le nombre de colonnes
print("Le tableau comporte {} colonne(s)".format(len(sous_nutrition.columns)))
print("Les types de donneés dans chaque colonne sont :\n{}".format(sous_nutrition.dtypes))
print("Pour chaque colonne, voici le nombre de valeurs :\n{}".format(sous_nutrition.count()))

Le tableau comporte 3 colonne(s)
Les types de donneés dans chaque colonne sont :
Zone      object
Année     object
Valeur    object
dtype: object
Pour chaque colonne, voici le nombre de valeurs :
Zone      1218
Année     1218
Valeur     624
dtype: int64


In [24]:
#Afficher les 5 premières lignes de la table
sous_nutrition.head()

,Zone,Année,Valeur
0,Afghanistan,2012-2014,8.6
1,Afghanistan,2013-2015,8.8
2,Afghanistan,2014-2016,8.9
3,Afghanistan,2015-2017,9.7
4,Afghanistan,2016-2018,10.5


In [25]:
#Conversion de la colonne sous nutrition en numérique
sous_nutrition['Valeur'] = sous_nutrition['Valeur'].replace('<0.1', None)
sous_nutrition['Valeur'] = sous_nutrition['Valeur'].astype(float)

In [26]:
#Conversion de la colonne (avec l'argument errors=coerce qui permet de convertir automatiquement les lignes qui ne sont pas des nombres en NaN)
#Puis remplacement des NaN en 0
sous_nutrition['Valeur'] = pd.to_numeric(sous_nutrition['Valeur'], errors='coerce')
sous_nutrition = sous_nutrition.fillna(0)
sous_nutrition.head()

,Zone,Année,Valeur
0,Afghanistan,2012-2014,8.6
1,Afghanistan,2013-2015,8.8
2,Afghanistan,2014-2016,8.9
3,Afghanistan,2015-2017,9.7
4,Afghanistan,2016-2018,10.5


In [27]:
#changement du nom de la colonne Valeur par sous_nutrition
sous_nutrition=sous_nutrition.rename(columns={'Valeur': 'sous_nutrition'})
sous_nutrition.head()

,Zone,Année,sous_nutrition
0,Afghanistan,2012-2014,8.6
1,Afghanistan,2013-2015,8.8
2,Afghanistan,2014-2016,8.9
3,Afghanistan,2015-2017,9.7
4,Afghanistan,2016-2018,10.5


In [28]:
#Multiplication de la colonne sous_nutrition par 1000000
sous_nutrition['sous_nutrition']=sous_nutrition['sous_nutrition']*1000000

In [29]:
#Afficher les 5 premières lignes de la table
sous_nutrition['sous_nutrition'] = sous_nutrition['sous_nutrition'].astype(int)
sous_nutrition.head()

,Zone,Année,sous_nutrition
0,Afghanistan,2012-2014,8600000
1,Afghanistan,2013-2015,8800000
2,Afghanistan,2014-2016,8900000
3,Afghanistan,2015-2017,9700000
4,Afghanistan,2016-2018,10500000


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.1 - Proportion de personnes en sous nutrition</h3>
</div>

In [30]:
# Il faut tout d'abord faire une jointure entre la table population et la table sous nutrition, en ciblant l'année 2017

#Je filtre les deux tables concernées pour l'année 2017
sous_nutrition_2017 = sous_nutrition[sous_nutrition['Année'] == '2016-2018']
population_2017 = population[population['Année'] == 2017]

#Je fais la fusion
pop_sous_nutri = pd.merge(sous_nutrition_2017, population_2017, on='Zone', how='inner')

In [31]:
#Affichage du dataset
pop_sous_nutri.head()

,Zone,Année_x,sous_nutrition,Année_y,Population
0,Afghanistan,2016-2018,10500000,2017,36296113
1,Afrique du Sud,2016-2018,3100000,2017,57009756
2,Albanie,2016-2018,100000,2017,2884169
3,Algérie,2016-2018,1300000,2017,41389189
4,Allemagne,2016-2018,0,2017,82658409


In [32]:
#Calcul et affichage du nombre de personnes en état de sous nutrition

#Je fais la somme des deux colonnes concernées 
total_pop_sous_nutri_2017 = pop_sous_nutri['sous_nutrition'].sum()
total_pop_2017 = pop_sous_nutri['Population'].sum()

#Je fais le calcul pour trouver la part des personnes sous-nutries
part_pop_sous_nutri_mondial = round(total_pop_sous_nutri_2017/total_pop_2017*100, 1)

print(f"Il y a dans le monde {total_pop_sous_nutri_2017:,} personnes en état de sous-nutrition. Cela représente {part_pop_sous_nutri_mondial} % de la population mondiale.")

Il y a dans le monde 535,700,000 personnes en état de sous-nutrition. Cela représente 7.1 % de la population mondiale.


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.2 - Nombre théorique de personne qui pourrait être nourries</h3>
</div>

In [33]:
#Combien mange en moyenne un être humain ? 

#Selon l'Oxfam qui relaie des chiffres des Etats-Unis, on peut retenir le chiffre de 2100 Kcal/jour par personne comme un seuil des besoins alimentaires essentiels.
#Source => https://www.oxfamfrance.org/agriculture-et-securite-alimentaire/faim-dans-le-monde-causes-consequences-solutions/#_ftnref3


In [34]:
#On commence par faire une jointure entre le data frame population et Dispo_alimentaire afin d'ajouter dans ce dernier la population
population_2017 = population[population['Année'] == 2017]
dispo_alim_pop_merge = pd.merge(dispo_alimentaire, population_2017[['Zone', 'Population']], on='Zone', how='inner')

In [35]:
#Affichage du nouveau dataframe  
dispo_alim_pop_merge.head()

,Zone,Produit,Origine,Disponibilité alimentaire (Kcal/personne/jour),Disponibilité alimentaire en quantité (kg/personne/an),Disponibilité de matière grasse en quantité (g/personne/jour),Disponibilité de protéines en quantité (g/personne/jour),Disponibilité intérieure,Exportations - Quantité,Importations - Quantité,Nourriture,Pertes,Production,Semences,Traitement,Aliments pour animaux,Autres Utilisations,Variation de stock,Population
0,Afghanistan,Abats Comestible,animale,5.0,1.72,0.20,0.77,53000000,0,0,53000000,0,53000000,0,0,0,0,0,36296113
1,Afghanistan,Agrumes- Autres,vegetale,1.0,1.29,0.01,0.02,41000000,2000000,40000000,39000000,2000000,3000000,0,0,0,0,0,36296113
2,Afghanistan,Aliments pour enfants,vegetale,1.0,0.06,0.01,0.03,2000000,0,2000000,2000000,0,0,0,0,0,0,0,36296113
3,Afghanistan,Ananas,vegetale,0.0,0.00,0.00,0.00,0,0,0,0,0,0,0,0,0,0,0,36296113
4,Afghanistan,Bananes,vegetale,4.0,2.70,0.02,0.05,82000000,0,82000000,82000000,0,0,0,0,0,0,0,36296113


In [36]:
#Création de la colonne dispo_kcal avec calcul des kcal disponibles mondialement
dispo_alim_pop_agrege = dispo_alim_pop_merge.pivot_table(
    index='Zone',
    aggfunc={
        'Disponibilité alimentaire (Kcal/personne/jour)': 'sum',
        'Population': 'first'
    }
)
dispo_alim_pop_agrege=dispo_alim_pop_agrege.rename(columns={'Disponibilité alimentaire (Kcal/personne/jour)': 'Disponibilité alimentaire total (Kcal/personne/jour)'})
dispo_alim_pop_agrege.head()

,Disponibilité alimentaire total (Kcal/personne/jour),Population
Zone,,
Afghanistan,2087.0,36296113
Afrique du Sud,3020.0,57009756
Albanie,3188.0,2884169
Algérie,3293.0,41389189
Allemagne,3503.0,82658409


In [37]:
#Calcul du nombre d'humains pouvant être nourris

#Rajout des deux colonnes 'Nombre de personnes pouvant être nourris' et 'Part de la population'
dispo_alim_pop_agrege['Nombre de personnes pouvant être nourris']=dispo_alim_pop_agrege['Disponibilité alimentaire total (Kcal/personne/jour)']*dispo_alim_pop_agrege['Population'] / 2100
dispo_alim_pop_agrege['Nombre de personnes pouvant être nourris']=dispo_alim_pop_agrege['Nombre de personnes pouvant être nourris'].astype(int)
dispo_alim_pop_agrege['Part de la population']=round(dispo_alim_pop_agrege['Nombre de personnes pouvant être nourris']/dispo_alim_pop_agrege['Population'] * 100,1)
dispo_alim_pop_agrege.head(5)

#Calcul du nombre de personnes à l'échelle mondiale
total_pop_mondiale=dispo_alim_pop_agrege['Population'].sum()
total_capacité_personnes_nourri = dispo_alim_pop_agrege['Nombre de personnes pouvant être nourris'].sum()
part_mondial_pers_nutri = round(total_capacité_personnes_nourri/total_pop_mondiale*100,1)

print(f"Au regard des capacités de production de nourriture, le nombre d'humains pouvant être nourris est de {total_capacité_personnes_nourri}, soit {part_mondial_pers_nutri} % de la population mondiale.")

Au regard des capacités de production de nourriture, le nombre d'humains pouvant être nourris est de 10086744073, soit 136.9 % de la population mondiale.


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.3 - Nombre théorique de personne qui pourrait être nourrie avec les produits végétaux</h3>
</div>

In [38]:
#Transfert des données avec les végétaux dans un nouveau dataframe
population_2017 = population[population['Année'] == 2017]
dispo_alimentaire_vegetaux=dispo_alimentaire[dispo_alimentaire['Origine'] == 'vegetale']
population_2017['Population'].astype(int)

dispo_alimentaire_vegetaux = dispo_alimentaire_vegetaux.drop(columns=[
    "Aliments pour animaux", 
    "Autres Utilisations", 
    "Disponibilité alimentaire en quantité (kg/personne/an)",
    "Disponibilité de matière grasse en quantité (g/personne/jour)",
    "Disponibilité de protéines en quantité (g/personne/jour)",
    "Disponibilité intérieure", "Exportations - Quantité", 
    "Importations - Quantité", 
    "Nourriture", 
    "Pertes",
    "Production",
    "Semences",
    "Traitement",
    "Variation de stock"])

pop_dispo_vegetaux_merge = pd.merge(dispo_alimentaire_vegetaux, population_2017, on = 'Zone', how='left')

display(pop_dispo_vegetaux_merge)

,Zone,Produit,Origine,Disponibilité alimentaire (Kcal/personne/jour),Année,Population
0,Afghanistan,Agrumes- Autres,vegetale,1.0,2017,36296113
1,Afghanistan,Aliments pour enfants,vegetale,1.0,2017,36296113
2,Afghanistan,Ananas,vegetale,0.0,2017,36296113
3,Afghanistan,Bananes,vegetale,4.0,2017,36296113
4,Afghanistan,Bière,vegetale,0.0,2017,36296113
...,...,...,...,...,...,...
11891,Îles Salomon,Sésame,vegetale,0.0,2017,636039
11892,Îles Salomon,Thé,vegetale,0.0,2017,636039
11893,Îles Salomon,Tomates,vegetale,0.0,2017,636039
11894,Îles Salomon,Vin,vegetale,0.0,2017,636039


In [39]:
#Calcul du nombre de kcal disponible pour les végétaux

pop_dispo_vegetaux_merge = pop_dispo_vegetaux_merge.pivot_table(
    index='Zone',
    aggfunc={
        'Disponibilité alimentaire (Kcal/personne/jour)': 'sum',
        'Population': 'first'
    })

display(pop_dispo_vegetaux_merge)

,Disponibilité alimentaire (Kcal/personne/jour),Population
Zone,,
Afghanistan,1871.0,36296113
Afrique du Sud,2533.0,57009756
Albanie,2203.0,2884169
Algérie,2915.0,41389189
Allemagne,2461.0,82658409
...,...,...
Émirats arabes unis,2718.0,9487203
Équateur,1732.0,16785361
États-Unis d'Amérique,2698.0,325084756


In [40]:
#Calcul du nombre d'humains pouvant être nourris avec les végétaux

#Rajout des deux colonnes 'Nombre de personnes pouvant être nourris' et 'Part de la population'
pop_dispo_vegetaux_merge['Nombre de personnes pouvant être nourris']=pop_dispo_vegetaux_merge['Disponibilité alimentaire (Kcal/personne/jour)']*pop_dispo_vegetaux_merge['Population'] / 2100
pop_dispo_vegetaux_merge['Nombre de personnes pouvant être nourris']=pop_dispo_vegetaux_merge['Nombre de personnes pouvant être nourris'].astype(int)
pop_dispo_vegetaux_merge['Part de la population']=round(pop_dispo_vegetaux_merge['Nombre de personnes pouvant être nourris']/pop_dispo_vegetaux_merge['Population'] * 100,1)

display(pop_dispo_vegetaux_merge.head(5))

#Calcul du nombre de personnes à l'échelle mondiale
total_pop_mondiale=dispo_alim_pop_agrege['Population'].sum()
total_capacité_personnes_nourri_vege = pop_dispo_vegetaux_merge['Nombre de personnes pouvant être nourris'].sum()
part_mondial_pers_nutri_vege = round(total_capacité_personnes_nourri_vege/total_pop_mondiale*100,1)

print(f"Au regard des capacités de production de nourriture, le nombre d'humains pouvant être nourris est de {total_capacité_personnes_nourri_vege}, soit {part_mondial_pers_nutri_vege} % de la population mondiale.")



,Disponibilité alimentaire (Kcal/personne/jour),Population,Nombre de personnes pouvant être nourris,Part de la population
Zone,,,,
Afghanistan,1871.0,36296113,32338108,89.1
Afrique du Sud,2533.0,57009756,68764624,120.6
Albanie,2203.0,2884169,3025630,104.9
Algérie,2915.0,41389189,57452136,138.8
Allemagne,2461.0,82658409,96867783,117.2


Au regard des capacités de production de nourriture, le nombre d'humains pouvant être nourris est de 8309290114, soit 112.8 % de la population mondiale.


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.4 - Utilisation de la disponibilité intérieure</h3>
</div>

In [41]:
#Calcul de la disponibilité totale

dispo_alim_mondiale = dispo_alimentaire.pivot_table(
    index ='Zone',
    values = 'Disponibilité intérieure',
    aggfunc ='sum'
)
display(dispo_alim_mondiale)
print(f"Le total de la disponibilité intérieure mondiale est de {dispo_alimentaire['Disponibilité intérieure'].sum():,} tonnes.")

,Disponibilité intérieure
Zone,
Afghanistan,13515000000
Afrique du Sud,61256000000
Albanie,4758000000
Algérie,42630000000
Allemagne,162275000000
...,...
Émirats arabes unis,11850000000
Équateur,18614000000
États-Unis d'Amérique,777992000000


Le total de la disponibilité intérieure mondiale est de 9,848,994,000,000 tonnes.


In [42]:
# Liste des colonnes qu'on veut analyser
colonnes_a_ajouter = ['Nourriture', 'Pertes', 'Semences', 'Traitement', 'Aliments pour animaux', 'Autres Utilisations' ]


# Étape 2 : pour chaque colonne dans notre liste (ex: 'Nourriture', 'Pertes')
for col in colonnes_a_ajouter:
    # On crée une table temporaire avec la somme de cette colonne par pays
    temp = dispo_alimentaire.pivot_table(
        index='Zone',
        values=col,
        aggfunc='sum'
    )

    # On crée une nouvelle colonne dans notre tableau principal avec le suffixe "_part"
    # Elle contient la part de cette catégorie (Nourriture, Pertes...) dans la dispo intérieure
    dispo_alim_mondiale[col + '_part'] = round(temp[col] / dispo_alim_mondiale['Disponibilité intérieure']*100,1)

display(dispo_alim_mondiale)

,Disponibilité intérieure,Nourriture_part,Pertes_part,Semences_part,Traitement_part,Aliments pour animaux_part,Autres Utilisations_part
Zone,,,,,,,
Afghanistan,13515000000,79.4,8.4,2.9,0.5,5.7,3.1
Afrique du Sud,61256000000,48.7,3.6,0.4,37.3,8.7,1.4
Albanie,4758000000,73.1,5.8,1.2,2.4,13.9,3.7
Algérie,42630000000,74.4,8.8,1.1,1.5,10.2,4.0
Allemagne,162275000000,48.8,2.3,1.0,24.6,18.6,4.8
...,...,...,...,...,...,...,...
Émirats arabes unis,11850000000,55.3,5.9,0.0,6.8,9.9,22.7
Équateur,18614000000,46.1,3.8,0.7,33.3,6.4,10.3
États-Unis d'Amérique,777992000000,41.2,0.9,1.3,18.1,19.1,19.9


In [43]:
#Calcul de la moyenne au niveau mondial
display(dispo_alim_mondiale['Nourriture_part'].mean())
display(dispo_alim_mondiale['Aliments pour animaux_part'].mean())
display(dispo_alim_mondiale['Pertes_part'].mean())
display(dispo_alim_mondiale['Semences_part'].mean())
display(dispo_alim_mondiale['Traitement_part'].mean())
display(dispo_alim_mondiale['Autres Utilisations_part'].mean())

57.517241379310356

12.026436781609195

4.6545977011494255

1.4741379310344827

17.02931034482759

7.470689655172413

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.5 - Utilisation des céréales</h3>
</div>

In [44]:
#Création d'une liste avec toutes les variables
df_cereales = dispo_alimentaire.loc[(dispo_alimentaire["Produit"] == "Millet") | (dispo_alimentaire["Produit"] == "Orge") | (dispo_alimentaire["Produit"] == "Seigle")|
                            (dispo_alimentaire["Produit"] == "Céréales, Autres")| (dispo_alimentaire["Produit"] == "Blé")| (dispo_alimentaire["Produit"] == "Maïs")|
                            (dispo_alimentaire["Produit"] == "Avoine") | (dispo_alimentaire["Produit"] == "Riz (Eq Blanchi)")| (dispo_alimentaire["Produit"] == "Sorgho")]

In [45]:
#Création d'un dataframe avec les informations uniquement pour ces céréales
display(df_cereales)

,Zone,Produit,Origine,Disponibilité alimentaire (Kcal/personne/jour),Disponibilité alimentaire en quantité (kg/personne/an),Disponibilité de matière grasse en quantité (g/personne/jour),Disponibilité de protéines en quantité (g/personne/jour),Disponibilité intérieure,Exportations - Quantité,Importations - Quantité,Nourriture,Pertes,Production,Semences,Traitement,Aliments pour animaux,Autres Utilisations,Variation de stock
7,Afghanistan,Blé,vegetale,1369.0,160.23,4.69,36.91,5992000000,0,1173000000,4895000000,775000000,5169000000,322000000,0,0,0,-350000000
32,Afghanistan,Maïs,vegetale,21.0,2.50,0.30,0.56,313000000,0,1000000,76000000,31000000,312000000,5000000,0,200000000,0,0
34,Afghanistan,Millet,vegetale,3.0,0.40,0.02,0.08,13000000,0,0,12000000,1000000,13000000,0,0,0,0,0
40,Afghanistan,Orge,vegetale,26.0,2.92,0.24,0.79,524000000,0,10000000,89000000,52000000,514000000,22000000,0,360000000,0,0
47,Afghanistan,Riz (Eq Blanchi),vegetale,141.0,13.82,0.27,2.70,461000000,0,119000000,422000000,24000000,342000000,14000000,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
15537,Îles Salomon,Blé,vegetale,184.0,25.37,1.00,5.19,14000000,0,14000000,14000000,0,0,0,0,0,0,0
15568,Îles Salomon,Maïs,vegetale,1.0,0.15,0.01,0.03,0,0,0,0,0,0,0,0,0,0,0
15575,Îles Salomon,Orge,vegetale,0.0,0.07,0.00,0.01,1000000,0,1000000,0,0,0,0,1000000,0,0,0
15591,Îles Salomon,Riz (Eq Blanchi),vegetale,623.0,63.76,1.36,10.90,49000000,0,47000000,36000000,1000000,3000000,0,0,0,12000000,0


In [46]:
#Affichage de la proportion d'alimentation animale
df_cereales_animal_pivot = df_cereales.pivot_table(index='Zone', values=['Disponibilité intérieure','Aliments pour animaux'], aggfunc='sum')

# Rajour d'une colonne indiquant la part par pays
df_cereales_animal_pivot['Part alimentation animale (%)'] = round(
    (df_cereales_animal_pivot['Aliments pour animaux'] / df_cereales_animal_pivot['Disponibilité intérieure']) * 100, 1)
display(df_cereales_animal_pivot)

# Calcul de la part mondial
part_mondial_animal = df_cereales_animal_pivot['Aliments pour animaux'].sum()/df_cereales_animal_pivot['Disponibilité intérieure'].sum()*100
print(f"La proportion des céréales consacrée à l'alimnentation animale au niveau mondial est de {round(part_mondial_animal,1)} %.")


,Aliments pour animaux,Disponibilité intérieure,Part alimentation animale (%)
Zone,,,
Afghanistan,560000000,7303000000,7.7
Afrique du Sud,4816000000,15481000000,31.1
Albanie,411000000,1145000000,35.9
Algérie,4335000000,15009000000,28.9
Allemagne,24242000000,39380000000,61.6
...,...,...,...
Émirats arabes unis,1052000000,3182000000,33.1
Équateur,846000000,3405000000,24.8
États-Unis d'Amérique,140019000000,342735000000,40.9


La proportion des céréales consacrée à l'alimnentation animale au niveau mondial est de 35.9 %.


In [47]:
#Affichage de la proportion d'alimentation humaine (correction consigne, il y avait deux fois alimentation animale)
df_cereales_humain_pivot = df_cereales.pivot_table(index='Zone', values=['Disponibilité intérieure','Nourriture'], aggfunc='sum')

# Rajour d'une colonne indiquant la part par pays
df_cereales_humain_pivot['Part alimetation humaine (%)'] = round(
    (df_cereales_humain_pivot['Nourriture'] / df_cereales_humain_pivot['Disponibilité intérieure']) * 100, 1)
display(df_cereales_humain_pivot)

# Calcul de la part mondial
part_mondial_humain = df_cereales_humain_pivot['Nourriture'].sum()/df_cereales_humain_pivot['Disponibilité intérieure'].sum()*100

print(f"La proportion des céréales consacrée à l'alimentation humaine au niveau mondial est de {round(part_mondial_humain,1)} %.")

,Disponibilité intérieure,Nourriture,Part alimetation humaine (%)
Zone,,,
Afghanistan,7303000000,5494000000,75.2
Afrique du Sud,15481000000,9499000000,61.4
Albanie,1145000000,474000000,41.4
Algérie,15009000000,8534000000,56.9
Allemagne,39380000000,9135000000,23.2
...,...,...,...
Émirats arabes unis,3182000000,1484000000,46.6
Équateur,3405000000,1504000000,44.2
États-Unis d'Amérique,342735000000,33610000000,9.8


La proportion des céréales consacrée à l'alimentation humaine au niveau mondial est de 43.0 %.


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.6 - Pays avec la proportion de personnes sous-alimentée la plus forte en 2017</h3>
</div>

In [48]:
#Création de la colonne proportion par pays
pop_sous_nutri['Part_des personnes_sous_nutries'] = (pop_sous_nutri['sous_nutrition']/pop_sous_nutri['Population']*100).round(1)
pop_sous_nutri = pop_sous_nutri.drop(columns=['Année_x', 'Année_y'])
pop_sous_nutri.head()

,Zone,sous_nutrition,Population,Part_des personnes_sous_nutries
0,Afghanistan,10500000,36296113,28.9
1,Afrique du Sud,3100000,57009756,5.4
2,Albanie,100000,2884169,3.5
3,Algérie,1300000,41389189,3.1
4,Allemagne,0,82658409,0.0


In [49]:
#affichage après trie des 10 pires pays
pop_sous_nutri.sort_values(by='Part_des personnes_sous_nutries', ascending=False).head(10)

,Zone,sous_nutrition,Population,Part_des personnes_sous_nutries
78,Haïti,5300000,10982366,48.3
157,République populaire démocratique de Corée,12000000,25429825,47.2
108,Madagascar,10500000,25570512,41.1
103,Libéria,1800000,4702226,38.3
100,Lesotho,800000,2091534,38.2
183,Tchad,5700000,15016753,38.0
161,Rwanda,4200000,11980961,35.1
121,Mozambique,9400000,28649018,32.8
186,Timor-Leste,400000,1243258,32.2
0,Afghanistan,10500000,36296113,28.9


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.7 - Pays qui ont le plus bénéficié d'aide alimentaire depuis 2013</h3>
</div>

In [50]:
#calcul du total de l'aide alimentaire par pays
aide_alim_total = aide_alimentaire.pivot_table(index = 'Zone', values = 'Valeur', aggfunc='sum')
display(aide_alim_total)

,Valeur
Zone,
Afghanistan,185452000
Algérie,81114000
Angola,5014000
Bangladesh,348188000
Bhoutan,2666000
...,...
Zambie,3026000
Zimbabwe,62570000
Égypte,1122000


In [51]:
#affichage après trie des 10 pays qui ont bénéficié le plus de l'aide alimentaire
aide_alim_total['Valeur'] = aide_alim_total['Valeur'] / 1_000_000
aide_alim_total.sort_values(by = 'Valeur', ascending=False).head(10)


,Valeur
Zone,
République arabe syrienne,1858.943
Éthiopie,1381.294
Yémen,1206.484
Soudan du Sud,695.248
Soudan,669.784
Kenya,552.836
Bangladesh,348.188
Somalie,292.678
République démocratique du Congo,288.502


In [52]:
print(f"L'aide alimentaire mondiale versée depuis 2016 s'élève à {round(aide_alim_total['Valeur'].sum(),1)} millions de tonnes")

L'aide alimentaire mondiale versée depuis 2016 s'élève à 11035.9 millions de tonnes


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.8 - Evolution des 5 pays qui ont le plus bénéficiés de l'aide alimentaire entre 2013 et 2016</h3>
</div>

In [56]:
#Création d'un dataframe avec la zone, l'année et l'aide alimentaire puis groupby sur zone et année 

aide_alim_zone_année = aide_alimentaire.groupby(['Zone', 'Année']).sum()
aide_alim_zone_année = aide_alim_zone_année.drop(columns=['Produit'])
aide_alim_zone_année.head(10)

Valeur
Zone        Année           
Afghanistan 2013   128238000
            2014    57214000
Algérie     2013    35234000
            2014    18980000
            2015    17424000
            2016     9476000
Angola      2013     5000000
            2014       14000
Bangladesh  2013   131018000
            2014   194628000

In [54]:
#Création d'une liste contenant les 5 pays qui ont le plus bénéficiées de l'aide alimentaire
top_5_aide_alim = aide_alim_zone_année.groupby(['Zone']).sum().sort_values(by = 'Valeur', ascending=False).head(5)
top_5_pays =['République arabe syrienne', 'Éthiopie', 'Yémen', 'Soudan du Sud', 'Soudan']
display(top_5_aide_alim)

,Valeur
Zone,
République arabe syrienne,1858943000
Éthiopie,1381294000
Yémen,1206484000
Soudan du Sud,695248000
Soudan,669784000


In [57]:
#On filtre sur le dataframe avec notre liste
aide_alim_zone_année = aide_alim_zone_année.reset_index()
aide_top_5_filtre = aide_alim_zone_année[aide_alim_zone_année['Zone'].isin(top_5_pays)]
display(aide_top_5_filtre)

,Zone,Année,Valeur
159,République arabe syrienne,2013,563566000
160,République arabe syrienne,2014,651870000
161,République arabe syrienne,2015,524949000
162,République arabe syrienne,2016,118558000
191,Soudan,2013,330230000
192,Soudan,2014,321904000
193,Soudan,2015,17650000
194,Soudan du Sud,2013,196330000
195,Soudan du Sud,2014,450610000
196,Soudan du Sud,2015,48308000


In [58]:
# Affichage des pays avec l'aide alimentaire par année
aide_top_5_filtre_pv = aide_top_5_filtre.pivot_table(index = 'Zone', columns = 'Année', values = 'Valeur', aggfunc='sum')
aide_top_5_filtre_pv = aide_top_5_filtre_pv.fillna(0).astype(int)
display(aide_top_5_filtre_pv)

Année,2013,2014,2015,2016
Zone,,,,
République arabe syrienne,563566000,651870000,524949000,118558000
Soudan,330230000,321904000,17650000,0
Soudan du Sud,196330000,450610000,48308000,0
Yémen,264764000,103840000,372306000,465574000
Éthiopie,591404000,586624000,203266000,0


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.9 - Pays avec le moins de disponibilité par habitant</h3>
</div>

In [59]:
#Calcul de la disponibilité en kcal par personne par jour par pays
dispo_alim_pop_merge_epure = dispo_alim_pop_merge.drop(columns = ['Produit', 'Origine', 'Disponibilité alimentaire en quantité (kg/personne/an)', 
                                     'Disponibilité de matière grasse en quantité (g/personne/jour)', 'Disponibilité de protéines en quantité (g/personne/jour)',
                                     'Disponibilité intérieure', 'Exportations - Quantité', 'Importations - Quantité', 'Nourriture', 'Pertes', 'Production',
                                     'Semences', 'Traitement', 'Aliments pour animaux', 'Autres Utilisations', 'Variation de stock', 'Population'
                                    ])
dispo_alim_par_pays = dispo_alim_pop_merge_epure.groupby('Zone').sum()
display(dispo_alim_par_pays)

,Disponibilité alimentaire (Kcal/personne/jour)
Zone,
Afghanistan,2087.0
Afrique du Sud,3020.0
Albanie,3188.0
Algérie,3293.0
Allemagne,3503.0
...,...
Émirats arabes unis,3275.0
Équateur,2346.0
États-Unis d'Amérique,3682.0


In [60]:
#Affichage des 10 pays qui ont le moins de dispo alimentaire par personne 
dispo_alim_par_pays.sort_values('Disponibilité alimentaire (Kcal/personne/jour)').head(10)

,Disponibilité alimentaire (Kcal/personne/jour)
Zone,
République centrafricaine,1879.0
Zambie,1924.0
Madagascar,2056.0
Afghanistan,2087.0
Haïti,2089.0
République populaire démocratique de Corée,2093.0
Tchad,2109.0
Zimbabwe,2113.0
Ouganda,2126.0


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.10 - Pays avec le plus de disponibilité par habitant</h3>
</div>

In [61]:
#Affichage des 10 pays qui ont le plus de dispo alimentaire par personne 
dispo_alim_par_pays.sort_values(by='Disponibilité alimentaire (Kcal/personne/jour)', ascending=False).head(10)

,Disponibilité alimentaire (Kcal/personne/jour)
Zone,
Autriche,3770.0
Belgique,3737.0
Turquie,3708.0
États-Unis d'Amérique,3682.0
Israël,3610.0
Irlande,3602.0
Italie,3578.0
Luxembourg,3540.0
Égypte,3518.0


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.11 - Exemple de la Thaïlande pour le Manioc</h3>
</div>

In [62]:
#création d'un dataframe avec uniquement la Thaïlande 
dispo_alim_thai_manioc = dispo_alimentaire[(dispo_alimentaire['Zone'] == 'Thaïlande')&
                                    (dispo_alimentaire['Produit'] == 'Manioc')
                                    ]
display(dispo_alim_thai_manioc)

,Zone,Produit,Origine,Disponibilité alimentaire (Kcal/personne/jour),Disponibilité alimentaire en quantité (kg/personne/an),Disponibilité de matière grasse en quantité (g/personne/jour),Disponibilité de protéines en quantité (g/personne/jour),Disponibilité intérieure,Exportations - Quantité,Importations - Quantité,Nourriture,Pertes,Production,Semences,Traitement,Aliments pour animaux,Autres Utilisations,Variation de stock
13809,Thaïlande,Manioc,vegetale,40.0,13.0,0.05,0.14,6264000000,25214000000,1250000000,871000000,1511000000,30228000000,0,0,1800000000,2081000000,0


In [63]:
#Calcul de la sous nutrition en Thaïlande

sous_nutri_thai_2017 = sous_nutrition[(sous_nutrition['Zone'] == 'Thaïlande')&
                (sous_nutrition['Année'] == '2016-2018')
                ].iloc[0]['sous_nutrition']

pop_thailande_2017 = population[(population['Zone'] == 'Thaïlande')&
                    (population['Année'] == 2017)
                    ].iloc[0]['Population']

print(f"La population  en état de sous-nutrition en Thaïlande est de {sous_nutri_thai_2017} personnes, soit {round((sous_nutri_thai_2017)/(pop_thailande_2017)*100,1)} % de sa population.")


La population  en état de sous-nutrition en Thaïlande est de 6200000 personnes, soit 9.0 % de sa population.


In [64]:
# On calcule la proportion exportée en fonction de la proportion (production ?)
part_exportation = dispo_alim_thai_manioc['Exportations - Quantité']/dispo_alim_thai_manioc['Production']*100
export_thai = round(part_exportation.iloc[0], 1)
print(f"La part de la production qui est exportée est de {export_thai} %.") 

La part de la production qui est exportée est de 83.4 %.


In [65]:
dispo_alim_thai = dispo_alimentaire[(dispo_alimentaire['Zone'] == 'Thaïlande')]
dispo_intérieure_total_thai = dispo_alim_thai['Disponibilité intérieure'].sum()
dispo_intérieure_kcal_total_thai = dispo_alim_thai['Disponibilité alimentaire (Kcal/personne/jour)'].sum()
dispo_intérieure_manioc_thai = dispo_alim_thai_manioc[dispo_alim_thai_manioc['Produit'] == 'Manioc'].iloc[0]['Disponibilité intérieure']

part_manioc_dispo_int_total_thai = dispo_intérieure_manioc_thai / dispo_intérieure_total_thai*100
print(f"La part du manioc disponible dans la disponibilité totale intérieure du pays est de {round(part_manioc_dispo_int_total_thai,1)} %.")

La part du manioc disponible dans la disponibilité toale intérieure du pays est de 4.0 %.


<div style="background-color: RGB(0,150,250);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">Etape 6 - Analyse complémentaires</h2>
</div>

<div style="background-color: RGB(0,250,250);" >
Analyse aide alimentaire par habitant

In [66]:
#J'importe un dataframe précédemment créé
aide_alim_zone_année = aide_alimentaire.groupby(['Zone', 'Année']).sum()
aide_alim_zone_année = aide_alim_zone_année.drop(columns=['Produit'])

#Je créé un dataframe pour le fichier population avec le même format
population_pays_année = population.groupby(['Zone', 'Année']).sum()

# Je fusionne les deux
population_aide_alimentaire_pays_année = pd.merge(aide_alim_zone_année, population_pays_année, on=['Zone','Année'], how='inner')

#Je rajoute une colonne avec la calcul du ration en kilo d'aide alimentaire par habitant
population_aide_alimentaire_pays_année['Aide par habitant en kg']=round(population_aide_alimentaire_pays_année['Valeur']/population_aide_alimentaire_pays_année['Population'],5)

#Suppression d'une valeur égale à 0
population_aide_alimentaire_pays_année = population_aide_alimentaire_pays_année[population_aide_alimentaire_pays_année['Valeur'] != 0]

population_aide_alimentaire_pays_année.head(5)

Valeur  Population  Aide par habitant en kg
Zone        Année                                                
Afghanistan 2013   128238000    32269589                  3.97396
            2014    57214000    33370794                  1.71449
Algérie     2013    35234000    38140133                  0.92380
            2014    18980000    38923692                  0.48762
            2015    17424000    39728025                  0.43858

In [67]:
population_aide_alimentaire_pays_année.sort_values(by = 'Aide par habitant en kg').head(10)

Valeur  Population  \
Zone                            Année                        
Angola                          2014     14000    26941779   
Bolivie (État plurinational de) 2014      6000    10706517   
Chine, continentale             2014   1298000  1399453965   
Cuba                            2014     14000    11306904   
Burkina Faso                    2016     72000    18646357   
Iran (République islamique d')  2015    492000    78492215   
                                2016    582000    79563989   
Géorgie                         2013     34000     4048701   
                                2014     36000     4035311   
Égypte                          2013   1122000    88404646   

                                       Aide par habitant en kg  
Zone                            Année                           
Angola                          2014                   0.00052  
Bolivie (État plurinational de) 2014                   0.00056  
Chine, continentale             2014                   0.00093  
Cuba                            2014                   0.00124  
Burkina Faso                    2016                   0.00386  
Iran (République islamique d')  2015                   0.00627  
                                2016                   0.00731  
Géorgie                         2013                   0.00840  
                                2014                   0.00892  
Égypte                          2013                   0.01269

In [68]:
population_aide_alimentaire_pays_année.sort_values(by = 'Aide par habitant en kg',ascending = False).head(10)


Valeur  Population  \
Zone                      Année                          
Soudan du Sud             2014   450610000    10554880   
République arabe syrienne 2014   651870000    18710711   
                          2015   524949000    17997408   
                          2013   563566000    19578461   
Djibouti                  2013    20368000      883293   
                          2014    17088000      898695   
Soudan du Sud             2013   196330000    10355035   
Palestine                 2014    83134000     4428639   
Yémen                     2016   465574000    27168208   
                          2015   372306000    26497889   

                                 Aide par habitant en kg  
Zone                      Année                           
Soudan du Sud             2014                  42.69210  
République arabe syrienne 2014                  34.83940  
                          2015                  29.16803  
                          2013                  28.78500  
Djibouti                  2013                  23.05917  
                          2014                  19.01424  
Soudan du Sud             2013                  18.95986  
Palestine                 2014                  18.77191  
Yémen                     2016                  17.13672  
                          2015                  14.05040